ejp_xml_pipeline/etl.py

```diff
@@ -80,23 +92,110 @@ def etl_ejp_xml_zip(
-                entity.table_name,
-                entity.get_full_file_location()
+            obj_key = get_temp_s3_object_name(
+                entity.s3_object_prefix,
+                original_obj_key
+            )
+            upload_file_into_s3(
+                bucket=ejp_xml_load_config.temp_file_s3_bucket,
+                object_key=obj_key,
+                full_file_path=entity.get_full_file_location()
             )
 
-            load_file_into_bq(
-                filename=entity.get_full_file_location(),
-                table_name=entity.table_name,
-                dataset_name=ejp_xml_load_config.dataset,
-                project_name=ejp_xml_load_config.gcp_project
+
+def load_entity_file_to_bq(
+        gcp_project: str,
+        dataset: str,
+        table_name: str,
+        file_path: str
+):
+    if os.path.getsize(file_path) > 0:
+        create_or_extend_table_schema(
+            gcp_project,
+            dataset,
+            table_name,
+            file_path
+        )
+        load_file_into_bq(
+            filename=file_path,
+            table_name=table_name,
+            dataset_name=dataset,
+            project_name=gcp_project
+        )
+
+
+# pylint: disable='too-many-arguments'
+def download_load2bq_cleanup_temp_files(
+        matching_file_metadata_iter, s3_bucket: str,
+        gcp_project: str, dataset: str,
+        bq_table: str, batch_size_limit: int = 5000
+):
+    written_file_row_count = 0
+    s3_objects_written_to_file = []
+    with NamedTemporaryFile() as named_temp_file:
+        tempfile_name = named_temp_file.name
+        with open(tempfile_name, 'a') as writer:
+            for matching_file_metadata, _ in matching_file_metadata_iter:
+                s3_object = matching_file_metadata.get(
+                    named_literals.S3_FILE_METADATA_NAME_KEY
+                )
+                jsonl_string = download_s3_object_as_string(
+                    s3_bucket,
+                    s3_object
+                )
+                writer.write(jsonl_string)
+                written_file_row_count += sum(
+                    map(lambda x: 1 if '\n' in x else 0, jsonl_string)
+                )
+                s3_objects_written_to_file.append(
+                    s3_object
+                )
+                if written_file_row_count > batch_size_limit:
+                    load_and_delete_temp_objects(
+                        gcp_project, dataset,
+                        bq_table, tempfile_name,
+                        s3_bucket, s3_objects_written_to_file
+                    )
+                    named_temp_file.truncate()
+                    s3_objects_written_to_file = []
+                    written_file_row_count = 0
+            load_and_delete_temp_objects(
+                gcp_project, dataset,
+                bq_table, tempfile_name,
+                s3_bucket, s3_objects_written_to_file
             )
+
+
+def load_and_delete_temp_objects(
```

Comment:
minor: just wondering whether there is much benefit of having this method. There doesn't seem to be any overlap in the parameters used.

Reply:
readability